identity-credential/src/credential/credential.rs

```diff
@@ -164,6 +164,43 @@ impl<T> Credential<T> {
   pub fn proof_mut(&mut self) -> Option<&mut Signature> {
     self.proof.as_mut()
   }
+
+  /// Checks whether this Credential does not expire before the given `Timestamp`.
+  /// True is returned in the case of no expiration date.
+  pub fn earliest_expiry_date(&self, timestamp: Timestamp) -> bool {
```

Comment:
Not sure about the naming here. Wouldn't names like "is_not_expired" and "is_past_issuance_time" (is_x) be a very fitting naming convention for the validation units?

Reply:
I think the current name better describes what the method actually checks. It checks that the credential expires no later than the supplied timestamp. In most use cases the timestamp would be now (and that is the default when running a `full_validation`), but for some one needs to check that the credential does not expire two weeks from now.